scipy/stats/_stats_py.py

```diff
@@ -63,6 +63,11 @@
 from scipy.optimize import root_scalar
 from scipy._lib.deprecation import _NoValue
 
+# In __all__ but deprecated for removal in SciPy 1.13.0
+from scipy._lib._util import float_factorial  # noqa
+from scipy.stats._mstats_basic import (PointbiserialrResult, Ttest_1sampResult,
+                                       Ttest_relResult)  # noqa
```

Comment:
There will be quite a lot of this. @h-vetinari if I understand correctly you expressed a preference for adding these imports rather than removing them from `__all__`, is this still the case?

Reply:
I guess you're referring to [this] comment? I have no problem with removing them from `__all__`, my point was that we should ideally have the test coverage that got added in #18649, but no preference how we do it.

In particular, we could write the test in a way to not depend on `__all__`, but to use a different way to ensure we don't accidentally remove them during the deprecation phase.

Does that make sense?

Reply:
> Does that make sense?

I don't understand the second paragraph.

IIRC, the test failure was legitimate - it failed due to one of the things it was trying to catch. The question was about whether the offending items should be "grandfathered out" (they had already been removed, so they should be allowed to stay removed) or whether they should be added back.

I think I have a preference for adding them back. Suppose the author made a mistake and added something that doesn't need to be added. No harm done. Suppose the author removes an item from `__all__` that wasn't supposed to be removed. A tiny bit of harm is done; the test no longer covers something it was supposed to cover. So for the sake of easy review I think it would be best to leave `__all__` alone for easy review.

Reply:
To me there are two unrelated questions: what should be in `__all__` and how do we test the deprecated functions don't get silently changed/removed (I'm happy the test caught something last PR, I'm not proposing to water that down).

For the latter, we could always adapt the tests correspondingly, which was my point. I have no strong preference on what should be in `__all__` or not, but I think it'd be preferable to not have deprecated functions in there, especially if it would otherwise cause a lot of input churn.

Reply:
> but I think it'd be preferable to not have deprecated functions in there, especially if it would otherwise cause a lot of input churn.

I am not sure we can do that otherwise we break `from <module> import *` users?

Reply:
Then just leave it in `__all__`. Again, I don't mind doing that at all, so to the degree that this thread started with what my supposed preference is: consider me neutral and continue with what you think is best.

Reply:
Apologies did not mean to misconstrue your opinion, just wanted to get to the bottom of what everyone thought on this